web_application/Backend/FlaskApp/app.py

```diff
@@ -623,101 +863,168 @@ def start_collaboration():
         data = request.get_json()
         collab_name = data.get('collabName')
         experiments = data.get('experiments', [])
-        raw_data = data.get('rawData')
-        meta_data = data.get('metaData')
         invited_users = data.get('invitedUsers', [])
+        creator_dataset_id = data.get('creatorDatasetId')
+        logging.info(invited_users)
 
         if not collab_name:
             logging.error("Collaboration name missing in the request")
             return jsonify({"error": "Collaboration name is required"}), 400
+            
 
         collaboration = {
             'uuid': str(uuid.uuid4()),
             'name': collab_name,
             'experiments': experiments,
-            'raw_data': raw_data,
-            'meta_data': meta_data,
             'creator_id': ObjectId(creator_id),
+            'creator_dataset_id': ObjectId(creator_dataset_id),
             'invited_users': [
                 {
                     'user_id': ObjectId(user['_id']),
+                    'user_dataset_id': ObjectId(user['dataset_id']),
                     'status': 'pending',
-                    'phenotype': user.get('phenotype')
+                    'phenotype': user.get('phenotype', 'N/A'),
                 } for user in invited_users
             ],
-            # 'created_at': datetime.datetime()
+            # 'created_at': datetime.datetime.utcnow()  # Optionally add timestamp
         }
 
         result = db.collaborations.insert_one(collaboration)
+        logging.info(f"Inserted collaboration with id: {result.inserted_id}")
+
 
         return jsonify({
             'message': 'Collaboration created successfully',
-            'collaboration_id': str(result.inserted_id),
             'collaboration_id': collaboration['uuid']
         }), 201
 
     except Exception as e:
         logging.error(f"Error creating collaboration: {str(e)}")
-        return jsonify({"error": str(e)}), 500
+        return jsonify({"error": "Internal server error"}), 500
 
-# Pending Development
-# @app.route('/api/collaboration/<uuid>', methods=['GET', 'POST'])
-# def collaboration_details(uuid):
-#     try:
-#         # Fetch collaboration details
-#         collaboration = db.collaborations.find_one({'uuid': uuid})
-#         if not collaboration:
-#             return jsonify({'error': 'Collaboration not found'}), 404
+@app.route('/api/collaboration/<uuid>', methods=['GET', 'POST'])
```

Comment:
I assume it be a GET. In terms of an update, it would be better to use a different function such as udpate_collaboration_details that accepts PUT. Combining GET and PUT will make the code hard to understand.